_includes/curation-workflow.md

```diff
@@ -0,0 +1,2164 @@
+option. Use -h to see the options. I started doing this as we found some
+cases where Gene Symbols had been submitted with trailing spaces and
+mapr counts them as different to Gene Symbols without the trailing
+space.
+
+IDR/idr-metadata/scripts/check\_csv\_columns.pl - does the same thing
+for the annotation.csv file but will not fix trailing spaces as these
+should be fixed in the files used to create the annotation.csv.
+
+IDR/idr-metadata/scripts/check\_utf8.py - checks whether a file contains
+valid UTF-8. Often the study file doesn’t with characters like degrees
+(º), pi (π), mu (µ) and beta (ß) in the protocols causing problems.
+This is not the best way to do it, but I usually just replace them with
+the closest “normal” character or word (o, pi, u and beta etc.) or
+remove them if it’s just something that has come out strangely.
+
+##### Bulkmap config file
+
+To create this file we need to go through every column in the
+annotation.csv and decide which ones should go into map annotations.
+Generally we want values in the map annotations that are searchable so
+we don’t include results values such as z-scores etc.
+
+There is both a “name” and “client name” e.g.
+
+\- name: Characteristics \[Strain\]
+
+clientname: Strain
+
+include: yes
+
+The “name” is the name of the column in the annotation.csv file.
+
+The “clientname” is the name you want it to have in the IDR map
+annotations.
+
+Some columns are put into particular mapr groups such as ‘organism’,
+‘gene’, ‘siRNA’. These ones are listed under particular mapr groups
+e.g.
+
+\- group:
+
+namespace: openmicroscopy.org/mapr/organism
+
+All the others are just listed but not under a mapr group (usually we
+list them first) and they appear in the ‘Others’ group in the Attributes
+section in the IDR.
+
+Look at previous bulkmap config files to see what the client names
+should be.
+
+If you want to add an external link for the value in another database
+add an extra line
+
+\- name: Gene Identifier
+
+clientname: Gene Identifier URL
+
+clientvalue: http://www.ncbi.nlm.nih.gov/gene/{{ value|urlencode }}
+
+include: yes
+
+Note that if you put a different URL for the same thing e.g. the same
+gene identifier, in two different studies you’ll end up with two
+different map annotations so you need to be careful about what has been
+used previously by checking through other bulkmap-config files.
+
+#### prodNN\_idr00XX\_input\_bulk.txt and prodNN\_idr00XX\_input.txt
+
+So that we can record what was annotated, and what was re-annotated in
+each release and to speed up the process of applying all the changes in
+idr-next Ola developed a method for adding bulk annotation files and the
+bulk-map annotation code using input files and shell scripts.
+
+The shell scripts themselves live in
+[<span class="underline">https://github.com/IDR/idr-metadata/tree/master/scripts/reannotate</span>](https://github.com/IDR/idr-metadata/tree/master/scripts/reannotate).
```

Comment:
Link is 404ing

Reply:
This is an old script and the instructions are out of date. I'll remove this paragraph